nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/test/java/org/apache/nifi/processors/splunk/TestPutSplunk.java

```diff
@@ -317,84 +260,39 @@ public void testUnableToCreateConnectionShouldRouteToFailure() {
+        if (protocol == TransportProtocol.UDP) {
+            createTestServer(address, NetworkUtils.getAvailableUdpPort(), protocol, sslContext);
+        } else {
+            createTestServer(address, NetworkUtils.getAvailableTcpPort(), protocol, sslContext);
         }
     }
 
-
-    /**
-     * A ChannelSender that captures each message that was sent.
-     */
-    private static class CapturingChannelSender extends ChannelSender {
-
-        private List<String> messages = new ArrayList<>();
-        private int count = 0;
-        private int errorStart = -1;
-        private int errorEnd = -1;
-
-        public CapturingChannelSender(String host, int port, int maxSendBufferSize, ComponentLog logger) {
-            super(host, port, maxSendBufferSize, logger);
-        }
-
-        @Override
-        public void open() throws IOException {
-
-        }
-
-        @Override
-        protected void write(byte[] data) throws IOException {
-            count++;
-            if (errorStart > 0 && count >= errorStart && errorEnd > 0 && count <= errorEnd) {
-                throw new IOException("this is an error");
-            }
-            messages.add(new String(data, StandardCharsets.UTF_8));
-        }
-
-        @Override
-        public boolean isConnected() {
-            return false;
-        }
-
-        @Override
-        public void close() {
-
-        }
-
-        public List<String> getMessages() {
-            return messages;
+    private void createTestServer(final String address, final int port, final TransportProtocol protocol, final SSLContext sslContext) {
+        messages = new LinkedBlockingQueue<>();
+        runner.setProperty(PutSplunk.PROTOCOL, protocol.name());
+        runner.setProperty(PutSplunk.PORT, String.valueOf(port));
+        final byte[] delimiter = OUTGOING_MESSAGE_DELIMITER.getBytes(CHARSET);
+        NettyEventServerFactory serverFactory = new ByteArrayMessageNettyEventServerFactory(runner.getLogger(), address, port, protocol, delimiter, VALID_LARGE_FILE_SIZE, messages);
+        if (sslContext != null) {
+            serverFactory.setSslContext(sslContext);
         }
+        eventServer = serverFactory.getEventServer();
+    }
 
-        public void setErrorStart(int errorStart) {
-            this.errorStart = errorStart;
+    private void checkReceivedAllData(final String[] sentData) throws Exception {
+        // check each sent FlowFile was successfully sent and received.
+        for (String item : sentData) {
+            ByteArrayMessage packet = messages.take();
+            assertNotNull(packet);
+            assertArrayEquals(item.getBytes(), packet.getMessage());
         }
 
-        public void setErrorEnd(int errorEnd) {
-            this.errorEnd = errorEnd;
-        }
+        // Check that we have no unexpected extra data.
+        assertNull(messages.poll());
```

Comment:
It might be helpful to add a message here, then the comment could be removed:
```suggestion
        assertNull("Unprocessed messages found", messages.poll());
```